mlir/lib/Catalyst/Transforms/DetectQNodes.cpp

```diff
@@ -284,29 +279,35 @@ void DetectQnodeTransform::rewrite(LLVM::CallOp callOp, PatternRewriter &rewrite
         PatternRewriter::InsertionGuard insertGuard(rewriter);
         rewriter.setInsertionPointToEnd(failBlock);
         rewriter.create<LLVM::UnreachableOp>(callOp->getLoc());
-        return;
     }
+    else {
 
-    auto successor = successBlock->getSuccessor(0);
+        auto successor = successBlock->getSuccessor(0);
+
+        // This is roughly what the function looks like after transformation
+        //     llvm.func caller() {
+        //         %0 = llvm.call @mlirAsyncRuntimeCreateToken() : () -> !llvm.ptr
+        //         %1 = llvm.call @mlirAsyncRuntimeCreateValue() : () -> !llvm.ptr
+        //         %null = llvm.mlir.null : !llvm.ptr
+        //         llvm.invoke @callee() to ^bbsuccess unwind ^bbfail : () -> ()
+        //     ^bbfail:
+        //         %3 = llvm.landingpad (catch %null : !llvm.ptr) : !llvm.struct<(ptr<i8>, i32)>
+        //         llvm.call @mlirAsyncRuntimeSetTokenError(%0) : (!llvm.ptr) -> ()
+        //         llvm.call @mlirAsyncRuntimeSetValueError(%1) : (!llvm.ptr) -> ()
+        //     ^bbsuccess:
+        //         llvm.call @mlirAsyncRuntimeEmplaceToken(%0) : (!llvm.ptr) -> ()
+        //         llvm.call @mlirAsyncRuntimeEmplaceValue(%1) : (!llvm.ptr) -> ()
+        //         llvm.br ^termination
+        //     ^termination:
+        //         // some cleanup and return
+        //     }
+        insertBranchFromFailToSuccessor(failBlock, successor, rewriter);
+    }
 
-    // This is roughly what the function looks like after transformation
-    //     llvm.func caller() {
-    //         %0 = llvm.call @mlirAsyncRuntimeCreateToken() : () -> !llvm.ptr
-    //         %1 = llvm.call @mlirAsyncRuntimeCreateValue() : () -> !llvm.ptr
-    //         %null = llvm.mlir.null : !llvm.ptr
-    //         llvm.invoke @callee() to ^bbsuccess unwind ^bbfail : () -> ()
-    //     ^bbfail:
-    //         %3 = llvm.landingpad (catch %null : !llvm.ptr) : !llvm.struct<(ptr<i8>, i32)>
-    //         llvm.call @mlirAsyncRuntimeSetTokenError(%0) : (!llvm.ptr) -> ()
-    //         llvm.call @mlirAsyncRuntimeSetValueError(%1) : (!llvm.ptr) -> ()
-    //     ^bbsuccess:
-    //         llvm.call @mlirAsyncRuntimeEmplaceToken(%0) : (!llvm.ptr) -> ()
-    //         llvm.call @mlirAsyncRuntimeEmplaceValue(%1) : (!llvm.ptr) -> ()
-    //         llvm.br ^termination
-    //     ^termination:
-    //         // some cleanup and return
-    //     }
-    insertBranchFromFailToSuccessor(failBlock, successor, rewriter);
+    // We will now annotate caller for the next stage.
+    //
+    //     llvm.func caller() attributes { catalyst.preHandleError }
+    scheduleAnalysisForErrorHandling(caller, rewriter);
```

Comment:
This additional instruction should be placed in a separate commit, as it is not related to "Documentation"

Reply:
I moved this function call from before the if statement to after the if statement. I just wanted the annotation for the next pass to be the last change in this function. This is also why I removed the return statement from the if statement.

Reply:
Thanks for clarifying. This clarification could have be a good commit description if it was made in a separate commit. As I am new at the code base, it is difficult for me to understand the motivation even for a minor change.